File: problems/easy/0088-merge-sorted-array/typescript/solution.ts
/**
 Do not return anything, modify nums1 in-place instead.
 */
export function merge(
  nums1: number[],
  m: number,
  nums2: number[],
  n: number,
): void {
  let p1 = m - 1; // Last element in nums1
  let p2 = n - 1; // Last element in nums2
  let p = m + n - 1; // Last position in nums1

  // Merge from back to front
  while (p1 >= 0 && p2 >= 0) {
    if (nums1[p1] > nums2[p2]) {
      nums1[p] = nums1[p1];
      p1--;
    } else {
      nums1[p] = nums2[p2];
      p2--;
    }
    p--;
  }

  // Copy remaining elements from nums2
  while (p2 >= 0) {
    nums1[p] = nums2[p2];
    p2--;
    p--;
  }
}

File: problems/easy/0088-merge-sorted-array/python/solution.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        end_of_nums1 = m - 1
        end_of_nums2 = n - 1
        last_position_of_num1 = m + n -1

        while end_of_nums1 >= 0 and end_of_nums2 >= 0:
            if(nums1[end_of_nums1] > nums2[end_of_nums2]):
                nums1[last_position_of_num1] = nums1[end_of_nums1]
                end_of_nums1 -= 1

            else:
                nums1[last_position_of_num1] = nums2[end_of_nums2]
                end_of_nums2 -= 1

            last_position_of_num1 -= 1


        while(end_of_nums2 >= 0):
            nums1[last_position_of_num1] = nums2[end_of_nums2]
            end_of_nums2 -= 1
            last_position_of_num1 -= 1